Installation issue: py-pynacl@1.5.0 with new installer

### Steps to reproduce the issue

**This error occurs with the new installer only; the package builds fine with the old installer.**

```console
$ spack spec -I py-pynacl@1.5.0
[+]  ie2ro65  py-pynacl@1.5.0 build_system=python_pip platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[+]  ipets45      ^compiler-wrapper@1.1.0 build_system=generic platform=linux os=almalinux9 target=skylake
[e]  u4dhsem      ^glibc@2.34 build_system=autotools platform=linux os=almalinux9 target=x86_64
[e]  gn6r5zv      ^intel-oneapi-compilers@2026.0.0~amd+envmods+fix_rt_linkage~nvidia build_system=generic platform=linux os=almalinux9 target=x86_64
[+]  bcf76fo      ^intel-oneapi-runtime@2026.0.0 build_system=generic platform=linux os=almalinux9 target=skylake
[+]  7dmki4l          ^gcc-runtime@11.5.0 build_system=generic platform=linux os=almalinux9 target=skylake
[e]  mgj3afl              ^gcc@11.5.0+binutils+bootstrap~graphite+libsanitizer~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++' platform=linux os=almalinux9 target=x86_64
[+]  2ku3egu          ^patchelf@0.17.2 build_system=autotools platform=linux os=almalinux9 target=skylake %c,cxx=gcc@11.5.0
[+]  vc3mfwe      ^py-cffi@1.17.1 build_system=python_pip platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[+]  c4hmqot          ^libffi@3.5.2 build_system=autotools platform=linux os=almalinux9 target=skylake %c,cxx=oneapi@2026.0.0
[+]  czlxdpc          ^pkg-config@0.29.2+internal_glib build_system=autotools platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[+]  ypliwgq          ^py-pycparser@2.21 build_system=python_pip platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[+]  vpezzi2      ^py-pip@26.1 build_system=generic platform=linux os=almalinux9 target=skylake
[+]  hen4hnu      ^py-setuptools@73.0.1 build_system=generic platform=linux os=almalinux9 target=skylake
[+]  whbs54r      ^py-wheel@0.45.1 build_system=generic platform=linux os=almalinux9 target=skylake
[+]  yfdlane      ^python@3.11.15+bz2~crypt+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~static~tests~tkinter+uuid+zlib build_system=generic patches:=13fa8bf,b0615b2,ebdca64,f2fd060 platform=linux os=almalinux9 target=skylake %c,cxx=oneapi@2026.0.0
[+]  vd6vutp          ^bzip2@1.0.8~debug~pic+shared build_system=generic platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[e]  nox4lnw              ^diffutils@3.7 build_system=autotools platform=linux os=almalinux9 target=x86_64
[+]  3v7j5z2          ^expat@2.8.1~libbsd build_system=autotools platform=linux os=almalinux9 target=skylake %c,cxx=oneapi@2026.0.0
[+]  je4c6ug          ^gdbm@1.26 build_system=autotools platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[e]  t7hfyzj          ^gettext@0.21+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools platform=linux os=almalinux9 target=x86_64
[+]  e5rxgb7          ^gmake@4.4.1~guile build_system=generic platform=linux os=almalinux9 target=skylake %c=gcc@11.5.0
[+]  i32pxjm          ^ncurses@6.6~symlinks+termlib abi=none build_system=autotools patches:=7a351bc platform=linux os=almalinux9 target=skylake %c,cxx=oneapi@2026.0.0
[+]  tv4a6n6          ^openssl@3.6.1~docs+shared build_system=generic certs=mozilla platform=linux os=almalinux9 target=skylake %c,cxx=oneapi@2026.0.0
[+]  bam62qt              ^ca-certificates-mozilla@2026-03-19 build_system=generic platform=linux os=almalinux9 target=skylake
[e]  pp6qa7l              ^perl@5.32.1~cpanm~opcode+open+shared+threads build_system=generic platform=linux os=almalinux9 target=x86_64
[+]  fecgjwc          ^readline@8.3 build_system=autotools patches:=21f0a03 platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[+]  3wqnf5o          ^sqlite@3.53.1+column_metadata+fts+rtree build_system=autotools platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[+]  ovdx34y          ^util-linux-uuid@2.41 build_system=autotools platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[+]  lkfm4y7          ^xz@5.8.3~pic build_system=autotools libs:=shared,static platform=linux os=almalinux9 target=skylake %c=oneapi@2026.0.0
[+]  7olbnpn          ^zlib@1.3.2+optimize+pic+shared build_system=makefile platform=linux os=almalinux9 target=skylake %c,cxx=oneapi@2026.0.0
[+]  cxuiujx      ^python-venv@1.0 build_system=generic platform=linux os=almalinux9 target=skylake
```


### Error message

```
==> The following package specs will be reinstalled:

-- linux-almalinux9-skylake / %c=oneapi@2026.0.0 ----------------
ie2ro65 py-pynacl@1.5.0 build_system=python_pip
==> Do you want to proceed? [y/N] y
[x] ie2ro65 py-pynacl@1.5.0 failed
-- lines 13 to 25 --
  Created temporary directory: /tmp/pip-ephem-wheel-cache-w3a91cld
  Processing ./.
    Added file:///home/dom/work/spack-stack/spack-stack-upd-spack-packages/cache/build_stage/spack-stage-py-pynacl-1.5.0-ie2ro65uugsrplloy6nlz6rqny5ygbhp/spack-src to build tracker '/tmp/pip-build-tracker-vqgfjvhc'
    Created temporary directory: /tmp/pip-modern-metadata-zn7nshta
    Preparing metadata (pyproject.toml): started
    Running command Preparing metadata (pyproject.toml)
>   /home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/py-setuptools-73.0.1-hen4hnu/lib/python3.11/site-packages/setuptools/_distutils/dist.py:268: UserWarning: Unknown distribution option: 'tests_require'
      warnings.warn(msg)
    running dist_info
    creating /tmp/pip-modern-metadata-zn7nshta/PyNaCl.egg-info
    writing /tmp/pip-modern-metadata-zn7nshta/PyNaCl.egg-info/PKG-INFO
    writing dependency_links to /tmp/pip-modern-metadata-zn7nshta/PyNaCl.egg-info/dependency_links.txt
    writing requirements to /tmp/pip-modern-metadata-zn7nshta/PyNaCl.egg-info/requires.txt
-- lines 44 to 56 --
    Removed PyNaCl==1.5.0 from file:///home/dom/work/spack-stack/spack-stack-upd-spack-packages/cache/build_stage/spack-stage-py-pynacl-1.5.0-ie2ro65uugsrplloy6nlz6rqny5ygbhp/spack-src from build tracker '/tmp/pip-build-tracker-vqgfjvhc'
  Created temporary directory: /tmp/pip-unpack-lzp4j0ah
  Building wheels for collected packages: PyNaCl
    Destination directory: /tmp/pip-ephem-wheel-cache-w3a91cld/wheels/f1/72/65/e90f45ebba4bbbeccbcda0cd5f2ac83e5dce8c4b52d64d6928/tmp7pcodkbz
    Building wheel for PyNaCl (pyproject.toml): started
    Running command Building wheel for PyNaCl (pyproject.toml)
>   /home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/py-setuptools-73.0.1-hen4hnu/lib/python3.11/site-packages/setuptools/_distutils/dist.py:268: UserWarning: Unknown distribution option: 'tests_require'
      warnings.warn(msg)
    running bdist_wheel
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-cpython-311
-- lines 319 to 331 --
    config.status: creating src/libsodium/include/Makefile
    config.status: creating src/libsodium/include/sodium/version.h
    config.status: creating test/default/Makefile
    config.status: creating test/Makefile
    config.status: executing depfiles commands
    config.status: executing libtool commands
>   make: *** internal error: invalid --jobserver-auth string 'fifo:/tmp/tmp7wr7iia4/jobserver_fifo'.  Stop.
    Traceback (most recent call last):
      File "/home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/py-pip-26.1-vpezzi2/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
        main()
      File "/home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/py-pip-26.1-vpezzi2/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 373, in main
        json_out["return_val"] = hook(**hook_input["kwargs"])
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-- lines 381 to 409 --
    ╰─> No available output.

    note: This error originates from a subprocess, and is likely not a problem with pip.
    full command: /home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/python-venv-1.0-cxuiujx/bin/python3 /home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/py-pip-26.1-vpezzi2/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py build_wheel /tmp/tmp8kx1fids
    cwd: /home/dom/work/spack-stack/spack-stack-upd-spack-packages/cache/build_stage/spack-stage-py-pynacl-1.5.0-ie2ro65uugsrplloy6nlz6rqny5ygbhp/spack-src
    Building wheel for PyNaCl (pyproject.toml): finished with status 'error'
>   ERROR: Failed building wheel for PyNaCl
  Failed to build PyNaCl
> error: failed-wheel-build-for-install

  × Failed to build installable wheels for some pyproject.toml based projects
  ╰─> PyNaCl
  Exception information:
  Traceback (most recent call last):
    File "/home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/py-pip-26.1-vpezzi2/lib/python3.11/site-packages/pip/_internal/cli/base_command.py", line 109, in _run_wrapper
      status = _inner_run()
               ^^^^^^^^^^^^
    File "/home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/py-pip-26.1-vpezzi2/lib/python3.11/site-packages/pip/_internal/cli/base_command.py", line 102, in _inner_run
      return self.run(options, args)
             ^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/py-pip-26.1-vpezzi2/lib/python3.11/site-packages/pip/_internal/cli/req_command.py", line 101, in wrapper
      return func(self, options, args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/py-pip-26.1-vpezzi2/lib/python3.11/site-packages/pip/_internal/commands/install.py", line 531, in run
      raise InstallWheelBuildError(build_failures)
  pip._internal.exceptions.InstallWheelBuildError: <InstallWheelBuildError: failed-wheel-build-for-install>
  Removed build tracker: '/tmp/pip-build-tracker-vqgfjvhc'
  Command exited with status 1:
      /home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/none/none/python-venv-1.0-cxuiujx/bin/python3 -m pip -vvv --no-input --no-cache-dir --disable-pip-version-check install --no-deps --ignore-installed --no-build-isolation --no-warn-script-location --no-index --prefix=/home/dom/work/spack-stack/spack-stack-upd-spack-packages/envs/ne-oneapi-2026.0.0/install/intel-oneapi-compilers/2026.0.0/py-pynacl-1.5.0-ie2ro65 .
==> Error: The following packages failed to install:
py-pynacl@1.5.0/ie2ro65uugsrplloy6nlz6rqny5ygbhp: /home/dom/work/spack-stack/spack-stack-upd-spack-packages/cache/build_stage/spack-stage-py-pynacl-1.5.0-ie2ro65uugsrplloy6nlz6rqny5ygbhp-lw3e_jmi.log
```

### Information on your system

```
* **Spack:** 1.2.0.dev0 (https://github.com/spack/spack/commit/6fb75bd0c607b5148c3a126301bc7a9e13b86b26)
* **Builtin repo:** /home/dom/work/spack-stack/spack-stack-upd-spack-packages/repos/builtin/repos/spack_repo/builtin (9505b92)
* **Python:** 3.9.25
* **Platform:** linux-almalinux9-skylake
```
This is a fork of spack-packages and spack; both were updated last week/this week from the authoritative repositories.


### Additional information

[spack-stage-py-pynacl-1.5.0-ie2ro65uugsrplloy6nlz6rqny5ygbhp-lw3e_jmi.log](https://github.com/user-attachments/files/28654337/spack-stage-py-pynacl-1.5.0-ie2ro65uugsrplloy6nlz6rqny5ygbhp-lw3e_jmi.log)

No maintainers for this package ... mentioning @haampie as the expert on the new installer.

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

```
>   make: *** internal error: invalid --jobserver-auth string 'fifo:/tmp/tmp7wr7iia4/jobserver_fifo'.  Stop.
```

The workaround is to add `depends_on("gmake", type="build")`.

If this is related to LTO, which I think it is, the actual issue is that GCC has a runtime dependency on `gmake`: it generates a makefile and runs make on it in `lto-wrapper`.

You can copy the solution from matplotlib (which presumably has a comment explaining it).

--- a/repos/spack_repo/builtin/packages/py_pynacl/package.py
+++ b/repos/spack_repo/builtin/packages/py_pynacl/package.py
@@ -17,7 +17,8 @@ class PyPynacl(PythonPackage):
     version("1.5.0", sha256="8ac7448f09ab85811607bdd21ec2464495ac8b7c66d146bf545b0f08fb9220ba")
     version("1.4.0", sha256="54e9a2c849c742006516ad56a88f5c74bf2ce92c9f67435187c3c5953b346505")
 
-    depends_on("c", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("gmake", type="build")
 
     depends_on("python@2.7:2.8,3.4:", type=("build", "run"))
     depends_on("python@3.6:", type=("build", "run"), when="@1.5.0:")

